### PR TITLE
NAV-241: Mock GPS Test

### PR DESCRIPTION
**Pull request by mfardinzaman**
Added bbb-eth-listener-test and nuc-eth-listener-test
Pulls off new mock GPS node published to by local pathfinding

### CODE REVIEW

**Review by hhenry01 — APPROVED** [review]

File: test/bbb_eth_listener_test/main.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+*  @brief Facilitates communication between the NUC and BBB 
+*         Network Table via ethernet connection
+*  
+*  Polls sensor data from the CANbus and publishes 
+*  it to the network table. Writes actuation angles
+*  outputted from the boat controller to the CANbus
+*  
+*  @author Alex Macdonald (Alexmac22347)
+*  @author Brielle Law (briellelaw)
+*  @author John Ahn (jahn18)
```
Comment (briellelaw) [inline test/bbb_eth_listener_test/main.cpp:15]: I'm gonna add Mo as an author here

File: test/bbb_eth_listener_test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Set a variable for commands below
```
Comment (briellelaw) [inline test/bbb_eth_listener_test/CMakeLists.txt:1]: I'm gonna try and move both the bbb_eth_listener & nuc_eth_listener test under the same directory

File: test/mock_gps_tests/MockGPSNucTest.cpp [hunk 3]
```diff
@@ -0,0 +1,279 @@
+// Copyright 2017 UBC Sailbot
+
+// COPIED FROM projects/nuc_eth_listener/main.cpp
```
Comment (briellelaw) [inline test/mock_gps_tests/MockGPSNucTest.cpp:3]: Thanks for this comment. Also gonna add doxygen style comment and Henry as an author

File: test/bbb_eth_listener_test/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# Setting up 
+Of course, make sure there is a connection between the BBB and the Intel NUC.
+This is typically done via ethernet cable, although technically it is not
+required (the devices could communicate over a home wifi network as well).
+Before running, the ethernet port must appear as an active network interface.
+You can view active interfaces on the BBB with  
+```ifconfig```
+
+# Running
+Make sure to run the bbb_eth_listener first, before running the nuc_eth_listener
```
Comment (briellelaw) [inline test/bbb_eth_listener_test/README.md:15]: Need to update to run bbb_eth_listener_test.
Will also update this README to note how it's being used to read mock gps data (and explain why we need to do this)

**Pull request closed without merging**